app/code/Magento/Theme/Block/Html/Title.php

```diff
@@ -27,6 +27,25 @@ class Title extends Template
     protected $pageTitle;
 
     /**
+     * Define if title should be translated, injected via layout xml
+     *
+     * @var bool
+     */
+    private $translateTitle;
+
+    /**
+     * Constructor
+     *
+     * @param Template\Context $context
+     * @param array $data
+     */
+    public function __construct(Template\Context $context, array $data = [])
+    {
+        $this->translateTitle = isset($data['translate_title']) ? (bool) $data['translate_title'] : true;
```

Comment:
Having flags in code to switch its behavior is almost never a good idea.
 
> category and product pages get their titles translated when they shouldn't

Why not simply introduce another title block implementation, `RawTitle` or something, and use it on mentioned pages?

General note: please always start contribution from `2.2-develop` branch, a default one in this repo.

Reply:
Noted